src/test/java/com/connexta/search/index/IndexServiceImplTest.java

```diff
@@ -6,239 +6,175 @@
-import java.util.function.Predicate;
 import java.util.stream.Stream;
-import okhttp3.mockwebserver.MockResponse;
-import okhttp3.mockwebserver.MockWebServer;
 import org.apache.commons.io.IOUtils;
+import org.apache.tika.exception.TikaException;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.MethodSource;
-import org.junit.jupiter.params.provider.ValueSource;
-import org.mockito.ArgumentCaptor;
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
-import org.springframework.core.io.Resource;
-import org.springframework.http.HttpStatus;
-import org.springframework.web.reactive.function.client.WebClient;
-import org.springframework.web.reactive.function.client.WebClient.RequestHeadersSpec;
-import org.springframework.web.reactive.function.client.WebClient.RequestHeadersUriSpec;
-import org.springframework.web.reactive.function.client.WebClient.ResponseSpec;
-import reactor.core.publisher.Mono;
 
 @ExtendWith(MockitoExtension.class)
 class IndexServiceImplTest {
 
-  @Mock private SearchManager mockSearchManager;
+  static final String DATASET_ID = "00067360b70e4acfab561fe593ad3f7a";
+  static final String FILE_URL = "http://file";
+  static final String IRM_URL = "http://irm";
 
-  @AfterEach
-  public void afterEach() {
-    verifyNoMoreInteractions(mockSearchManager);
-  }
+  final BodyContentExtractor mockBodyContentExtractor = mock(BodyContentExtractor.class);
+  final IndexRepository mockIndexRepository = mock(IndexRepository.class);
+  final IonResourceLoader mockGetInputStream = mock(IonResourceLoader.class);
 
-  @ParameterizedTest
-  @ValueSource(ints = {400, 401, 403, 500, 501})
-  void testIrmUriReturnsNotOkStatusCode(final int code) throws Exception {
-    final String datasetId = "00067360b70e4acfab561fe593ad3f7a";
+  final IndexService indexService =
+      new IndexServiceImpl(mockIndexRepository, mockGetInputStream, mockBodyContentExtractor);
 
-    final SearchException thrown;
-    try (final MockWebServer storeMockWebServer = new MockWebServer()) {
-      storeMockWebServer.start();
-      storeMockWebServer.enqueue(new MockResponse().setResponseCode(code));
-
-      thrown =
-          assertThrows(
-              SearchException.class,
-              () ->
-                  ((IndexService) new IndexServiceImpl(mockSearchManager, WebClient.create()))
-                      .index(
-                          datasetId,
-                          storeMockWebServer
-                              .url(String.format("/dataset/%s/irm", datasetId))
-                              .uri()));
-    }
-    assertThat(thrown.getStatus(), is(HttpStatus.BAD_REQUEST));
+  @AfterEach
+  void afterEach() {
+    reset(mockBodyContentExtractor, mockIndexRepository, mockGetInputStream);
```

Comment:
✏️ I think it's simpler and more readable to use mockito's annotation for creating mocks:

```java
@Mock private BodyContentExtractor mockBodyContentExtractor;
@Mock private IndexRepository mockIndexRepository;
@Mock private IonResourceLoader mockIonResourceLoader;

private IndexService indexService;

@BeforeEach
void beforeEach() {
  indexService = new IndexServiceImpl(mockIndexRepository, mockGetInputStream, mockBodyContentExtractor);
}
```